Make into jar

td-extreme opened the pull request.

td-extreme committed:
An instance of HttpServer can now be created.

td-extreme committed:
Update README.md

Update README.md

Update README.md

Update README.md

Update README.md

dadarek commented:
Here is a good opportunity to practice `git rebase -i HEAD~4` and get rid of those 3 `Update README.md` commits

Once you run that command the instructions that come up should be _pretty_ self-explanatory, but beware and _do backup_ your stuff :)

dadarek commented on src/main/java/server/HttpServer.java:
```diff
@@ -1,30 +1,45 @@
     catch (IOException e) {
-      System.out.println("\nPort number " + arguments.getPortNumber() + " can not be opened. It may be in use by another application.\n");
+      System.out.println("\nPort number " + portNumber + " can not be opened. It may be in use by another application.\n");
       return;
```
So this is a little shady and _probably_ want to avoid early returns from a constructor, especially when "something went wrong". If you return (and not blow up) then you'll have a "partially" constructed object. Now what the heck do you do with a partially constructed object?

dadarek commented on src/main/java/server/HttpServer.java:
```diff
@@ -1,30 +1,45 @@
+  }
+
+  public void run() {
+    httpServerRunner.run();
```
Is this method blocking? Will the following two statements get printed out only when the server stops (and returns from `run`), put differently never?

td-extreme replied:
Yeah, this should be below the other 2 statements

dadarek commented on src/main/java/server/HttpServer.java:
```diff
@@ -1,30 +1,45 @@
+    customHandlerRouter.addRoute(method, path, handler);
+  }
+
+  public static void main(String[] args) {
```
Everything is beautiful, now let's get main the heck out of here. Maybe put it into Main so one can Main.main.

dadarek commented on README.md:
```diff
@@ -29,6 +29,81 @@ gradle run -P args=”-p 1234 -d ./site”
+After you have created your own custom handler you can add a route to this handler by using the HttpServer’s method `addRoute(String path, String method, IHanlder handler)`
+
+Example of how to create an instance of the HttpServer, add a route to your custom handler and start the server.
+~~~~
```
markdown tip:

you can use three backticks on a line on their own, put some code, then put three more backticks, and github will format them nicely for you.

You can even tell it that the following code is "java" or "bash" or whatever.

Here you go buddy
https://guides.github.com/features/mastering-markdown/#GitHub-flavored-markdown

dadarek commented on README.md:
```diff
@@ -29,6 +29,81 @@ gradle run -P args=”-p 1234 -d ./site”
+An example of what a custom handler might look like.
+
+~~~~
+Public class CustomHandler implements IHanlder {
```
lowercase "P"

dadarek replied:
also Handler not Hanlder

mikeknep commented on README.md:
```diff
@@ -29,6 +29,81 @@ gradle run -P args=”-p 1234 -d ./site”
+~~~~
+Public class CustomHandler implements IHanlder {
+  public HttpResposne generateResposne() {
+    HttpResposne myResponse = new HttpResponse();
```
I see someone has been taking advantage of the Java compiler... Typos here and the line above.

mikeknep commented:
Really nice README

mikeknep commented:
Test in the other repo looks good. Very nice.

td-extreme committed:
Move main() out of Server and into sepearte class

dadarek commented:
:shipit:

td-extreme committed:
added additional codes

td-extreme committed:
Refactor to remove repetative text

dadarek commented:
Should this get merged?

td-extreme commented:
I still need to fix the problem with the early return in the constructor. I suppose I could go ahead and merge it and then create a story in artisan addressing that issue.

dadarek commented:
Oh yeah. No, go ahead and fix it here. I thought you went on to the Clojure story so I assumed this story was done.

td-extreme committed:
Added MethodNotAllowedHandler, PortInUseException

Removed the early return from HttpServer constructor, the run method now
will throw and UnableToOpenPortException if port is already in use.

Request Verifier no longer throws a invalid request exception if method
is not supported. Instead the handler rounter will select the
HandlerMethodNotAllowed.

dadarek commented on src/main/java/httpResponse/HttpResponseWriter.java:
```diff
@@ -42,7 +43,13 @@ private void writeToByteArrayOS(ByteArrayOutputStream os, byte[] bytesToWrite) {
 
   private String responseLine(int responseCode) {
-    return responseLines.get(responseCode);
+    StringBuilder returnString = new StringBuilder();
```
Yes!! I was waiting for this for 7 weeks!

td-extreme committed:
Fixed response code bug.

Fixed bug where all 5** codes were being added as 500 codes.

Made it so that if code is not in the list description returns empty
string instead of NULL.

td-extreme committed:
Removed splatted of import for java.io.*

Changed this to only import the two things needed forom java.io.

td-extreme committed:
Remvoe no longer need include of java.util.*

td-extreme committed:
Updated name for jar file and changed version #

Due to chagnes made I felt it would be wise to update the version number.

td-extreme committed:
Added internal server error

IOExceptions for file access request now return internal server error
codes.

td-extreme committed:
Fixed test to check for code 500 for internal err
